lib/resturant.rb

```diff
@@ -0,0 +1,42 @@
+require 'json'
+
+class Resturant
+
+  def initialize(menu = Menu, order_template = Order, printer = Printer, messenger = MessengerService)
+    @menu = menu.new.dishes
+    @order_template = order_template
+    @printer = printer.new
+    @messenger = messenger.new
+  end
+
+  def show_menu
+    JSON.pretty_generate menu
+  end
+
+  def add_to_cart(dish_info)
+    fail "Please provide a hash with ':name' and ':quantity' as keys" if dish_info[:name] == nil || dish_info[:quantity] == nil
+    @current_order ||= order_template.new
+    item = menu.select{|item| item.has_value?(dish_info[:name])}.first
```

Comment:
Could the menu have a method that finds and returns an item? This would simplify the `add_to_cart` method.